app/controllers/sections/documents/DocumentsAddToListController.scala

```diff
@@ -0,0 +1,93 @@
+import play.api.i18n.MessagesApi
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import services.UserAnswersService
+import viewmodels.helpers.DocumentsAddToListHelper
+import viewmodels.taskList.InProgress
+import views.html.sections.documents.DocumentsAddToListView
+
+import javax.inject.Inject
+import scala.concurrent.Future
+
+class DocumentsAddToListController @Inject()(
+                                       override val messagesApi: MessagesApi,
+                                       override val userAnswersService: UserAnswersService,
+                                       override val userAllowList: UserAllowListAction,
+                                       override val navigator: DocumentsNavigator,
+                                       override val auth: AuthAction,
+                                       override val getData: DataRetrievalAction,
+                                       override val requireData: DataRequiredAction,
+                                       formProvider: DocumentsAddToListFormProvider,
+                                       val controllerComponents: MessagesControllerComponents,
+                                       view: DocumentsAddToListView,
+                                       addToListHelper: DocumentsAddToListHelper
+                                     ) extends BaseNavigationController with AuthActionHelper {
+
+  def onPageLoad(ern: String, draftId: String): Action[AnyContent] =
+    authorisedDataRequest(ern, draftId) { implicit request =>
+      renderView(Ok, fillForm(DocumentsAddToListPage, formProvider()))
+    }
+
+  def onSubmit(ern: String, draftId: String): Action[AnyContent] =
+    authorisedDataRequestAsync(ern, draftId) { implicit request =>
+      formProvider().bindFromRequest().fold(
+        formWithErrors => Future(renderView(BadRequest, formWithErrors)),
+        handleSubmissionRedirect(_)
+      )
+    }
+
+  private def renderView(status: Status, form: Form[_])(implicit request: DataRequest[_]): Result = {
+    status(view(
+      form = form,
+      onSubmitCall = controllers.sections.documents.routes.DocumentsAddToListController.onSubmit(request.ern, request.draftId),
+      documents = addToListHelper.allDocumentsSummary(),
+      showNoOption = DocumentsSectionUnits.status != InProgress
+    ))
+  }
+
+  private def handleSubmissionRedirect(answer: DocumentsAddToList)(implicit request: DataRequest[_]): Future[Result] = {
+    answer match {
+      case DocumentsAddToList.Yes =>
+
+        val answersWithDocumentAddToList = request.userAnswers
+          .set(DocumentsAddToListPage, DocumentsAddToList.Yes)
+
+        userAnswersService.set(request.userAnswers.remove(DocumentsAddToListPage))
+          .map { _ =>
+            Redirect(navigator.nextPage(
+              page = DocumentsAddToListPage,
+              mode = NormalMode,
+              userAnswers = answersWithDocumentAddToList
+            ))
+          }
```

Comment:
I'm a little confused by this, why does it set this to `Yes` but then remove it when it stores it?

Reply:
Makes sense 👍